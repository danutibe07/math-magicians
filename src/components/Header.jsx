import React from 'react';
import Navbar from './Navbar';

const Header = () => (
  <div className="maths-header">
    <h2>Maths Magicians</h2>
    <Navbar />
  </div>
);

export default Header;
