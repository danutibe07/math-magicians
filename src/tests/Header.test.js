import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

it('test Header rendering', () => {
  render(
    <Router>
      <Header />
    </Router>,
  );
  const tree = renderer
    .create(
      <Router>
        <Header />
      </Router>,
    )
    .toJSON();

  const navbars = screen.getAllByRole('link');
  expect(navbars).toHaveLength(3);
  expect(tree).toMatchSnapshot();
});
