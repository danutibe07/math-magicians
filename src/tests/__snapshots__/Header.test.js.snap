// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Header rendering 1`] = `
<div
  className="maths-header"
>
  <h2>
    Maths Magicians
  </h2>
  <nav
    className="maths-navbar"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className=""
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      className=""
      href="/quotes"
      onClick={[Function]}
    >
      Quote
    </a>
  </nav>
</div>
`;
